rotkehlchen/chain/ethereum/modules/balancer/utils.py

```diff
@@ -0,0 +1,284 @@
+            token_weight = deserialize_asset_amount(raw_token['denormWeight'])
+        except KeyError as e:
+            raise DeserializationError(f'Missing key: {str(e)}.') from e
+
+        try:
+            token_address = to_checksum_address(token_address)
+        except ValueError as e:
+            raise DeserializationError(
+                f'Invalid ethereum address: {token_address} in pool token: {token_symbol}.',  # noqa: E501
+            ) from e
+
+        token = get_ethereum_token(
+            symbol=token_symbol,
+            ethereum_address=token_address,
+            name=token_name,
+            decimals=token_decimals,
+        )
+        token_user_amount = user_amount / total_amount * token_total_amount
+        weight = token_weight * 100 / total_weight
+        pool_token = BalancerPoolToken(
+            token=token,
+            total_amount=token_total_amount,
+            user_balance=Balance(amount=token_user_amount),
+            weight=weight,
+        )
+        pool_tokens.append(pool_token)
+
+    pool = BalancerPool(
+        address=address,
+        tokens=pool_tokens,
+        total_amount=total_amount,
+        user_balance=Balance(amount=user_amount),
+    )
+    return user_address, pool
+
+
+def deserialize_transaction_id(raw_tx_id: str) -> Tuple[str, int]:
+    try:
+        tx_hash, raw_log_index = raw_tx_id.split('-')
+        log_index = int(raw_log_index)
+    except ValueError as e:
+        raise DeserializationError(f'Unexpected transaction id: {raw_tx_id}') from e
+    return tx_hash, log_index
+
+
+def deserialize_swap(raw_swap: Dict[str, Any]) -> AMMSwap:
+    """May raise DeserializationError"""
+    attr_raw_addresses: Dict[str, str] = {}  # not checksum addresses
+    attr_addresses: Dict[str, ChecksumEthAddress] = {}  # checksum addresses
+    try:
+        tx_hash, log_index = deserialize_transaction_id(raw_swap['id'])
+        timestamp = deserialize_timestamp(raw_swap['timestamp'])
+        raw_tokens = raw_swap['poolAddress']['tokens']
+        token0_symbol = raw_swap['tokenInSym']
+        token1_symbol = raw_swap['tokenOutSym']
+        amount0_in = deserialize_asset_amount(raw_swap['tokenAmountIn'])
+        amount1_out = deserialize_asset_amount(raw_swap['tokenAmountOut'])
+        attr_raw_addresses['userAddress'] = raw_swap['userAddress']['id']  # address
+        attr_raw_addresses['caller'] = raw_swap['caller']  # from_address
+        attr_raw_addresses['poolAddress'] = raw_swap['poolAddress']['id']  # to_address
+        attr_raw_addresses['tokenIn'] = raw_swap['tokenIn']  # token0_address
+        attr_raw_addresses['tokenOut'] = raw_swap['tokenOut']  # token1_address
+    except KeyError as e:
+        raise DeserializationError(f'Missing key: {str(e)}.') from e
+
+    # Checksum addresses
+    for attr, raw_address in attr_raw_addresses.items():
+        try:
+            attr_addresses[attr] = to_checksum_address(raw_address)
```

Comment:
why not make this in a named tuple?

Reply:
The idea behind was to provide the exact key (name) of the unprocessable address, but I understand the concern about using dicts. I've already addressed it for using nametuple on bot groups of addresses (raw and checksum), but also being able to provide the exact field (python attr name) and address that is failing